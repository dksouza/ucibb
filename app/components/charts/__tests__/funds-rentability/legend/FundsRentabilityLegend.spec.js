import React from 'react';
import { shallow } from 'enzyme';
import FundPercentage from '../../../funds-rentability/legend';

let wrapper = {};
let props = {};

const data = [
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
    key: 'MULTI DINAMICO V',
    svg: { fill: '#005AA5' },
    hexColor: '#005AA5',
    percent: 9.09,
  },
];

describe('FundPercentage', () => {
  beforeAll(() => {
    props = {
      data,
    };

    return props;
  });

  beforeEach(() => {
    wrapper = shallow(<FundPercentage {...props} />);
    return wrapper;
  });

  it('Shallow View', () => {
    expect(wrapper.find('View').exists()).toBe(true);
  });

  it('Shallow FundsRentabilityLegendItem', () => {
    expect(wrapper.find('FundsRentabilityLegendItem').exists()).toBe(true);
  });

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
