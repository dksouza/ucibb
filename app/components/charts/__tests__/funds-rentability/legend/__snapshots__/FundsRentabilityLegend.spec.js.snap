// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundPercentage Snapshot 1`] = `
<View>
  <FundsRentabilityLegendItem
    accessibilityLabel="Item descritivo do grÃ¡fico, index 0"
    fundName="MULTI DINAMICO V"
    index={0}
    key="0"
    percent={9.09}
    value={277.3}
    yieldValue={0}
  />
</View>
`;
